src/it/scala/org/apache/spark/sql/snowflake/SFTestWrapperSparkSession.scala

```diff
@@ -0,0 +1,37 @@
+package org.apache.spark.sql.snowflake
+
+import net.snowflake.spark.snowflake.Utils
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import scala.reflect.runtime.universe.TypeTag
+
+class SFTestWrapperSparkSession(sc: SparkContext, sfConfigs: Map[String, String])
+    extends SparkSession(sc) {
+
+  override def createDataFrame[A <: Product: TypeTag](data: Seq[A]): DataFrame = {
+    // Create a DataFrame from the Seq using Spark's normal way.
+    val origDF = super.createDataFrame(data)
+
+    // Write this DataFrame to Snowflake, in effect testing the connector's write-out
+    // code path, and then read it back as a Snowflake DataFrame. This uses
+    // the temporary schema defined in the run properties, tempTestSchema
+    val tableName = s"tempTestTable_${Utils.randomSuffix}"
```

Comment:
This is nice, but your tests could take a long time to run. Consider caching the table name based on data, you are likely to see a lot of repeated use of the same sequence.

Reply:
Hmm, compared to copying the data over to Snowflake, I don't think generating a random string is going to use up all that much time. Admittedly, the write() and read() add a bunch of time to the testing, and this happens when the Dataframe is accessed for the first time in a suite. I don't have any other bright ideas for optimizing this.